crates/fluvio-spu/src/services/public/mod.rs

```diff
@@ -1,30 +1,143 @@
 mod api_versions;
-mod service_impl;
 mod produce_handler;
 mod fetch_handler;
 mod offset_request;
+mod offset_update;
 mod stream_fetch;
 
-use tracing::info;
+use std::sync::Arc;
+use async_trait::async_trait;
+use tracing::{info, debug, trace, instrument};
+use futures_util::StreamExt;
 
-use fluvio_service::FluvioApiServer;
-use service_impl::PublicService;
+use fluvio_socket::{FluvioSocket, SocketError};
+use fluvio_service::{FluvioApiServer, FluvioService, ConnectInfo, call_service};
 use fluvio_spu_schema::server::SpuServerRequest;
 use fluvio_spu_schema::server::SpuServerApiKey;
+use fluvio_types::event::StickyEvent;
 
 use crate::core::DefaultSharedGlobalContext;
+use self::api_versions::handle_api_version_request;
+use self::produce_handler::handle_produce_request;
+use self::fetch_handler::handle_fetch_request;
+use self::offset_request::handle_offset_request;
+use self::offset_update::handle_offset_update;
+use self::stream_fetch::StreamFetchHandler;
 pub use stream_fetch::publishers::StreamPublishers;
 
-pub(crate) type PublicApiServer =
-    FluvioApiServer<SpuServerRequest, SpuServerApiKey, DefaultSharedGlobalContext, PublicService>;
+pub(crate) type SpuPublicServer = FluvioApiServer<
+    SpuServerRequest,
+    SpuServerApiKey,
+    DefaultSharedGlobalContext,
+    SpuPublicService,
+>;
 
-// start server
-pub fn create_public_server(addr: String, ctx: DefaultSharedGlobalContext) -> PublicApiServer {
-    info!(
-        "starting SPU: {} at public service at: {}",
-        ctx.local_spu_id(),
-        addr
-    );
+#[derive(Debug)]
+pub struct SpuPublicService {
+    _0: (), // Prevent construction
```

Comment:
what's point of doing this?

Reply:
I don't really like having structs that have no fields, it seems like we should either have a unit struct (which can be constructed out of thin air), or to add one private field to prevent it from being constructed except via the `new` function. This is more future-proof, since it prevents destructuring on the type. There is no downside to doing this.